src/Controller/CRUDController.php

```diff
@@ -70,6 +70,8 @@ class CRUDController implements ContainerAwareInterface
     /**
      * The template registry of the related Admin class.
      *
+     * @deprecated since sonata-project/admin-bundle 3.x, to be removed in 4.0. Use `$this->admin->getTemplateRegistry()` instead.
+     *
      * @var TemplateRegistryInterface
      */
     private $templateRegistry;
```

Comment:
CRUDController has not own template registry. This is from admin. Remove it and use registry directly from admin will be more logic.